apps/devhub/tests/test_views.py

```diff
@@ -1207,6 +1212,23 @@ def test_delete_and_recreate_credentials(self):
         assert new_key.key != old_key.key
         assert new_key.secret != old_key.secret
 
+    def test_retreive_credentials(self):
+        client = Client(enforce_csrf_checks=True)
+        assert client.login(username='regular@mozilla.com', password='password')
+
+        response = client.get(reverse('devhub.api_creds'),
+                                   {'key': 'abc123'})
+        assert response.status_code == 200
+        response_json = json.loads(response.content)
+        assert response_json['secret'] == 'supersekret'
+
+    def test_retreive_credentials_404(self):
+        client = Client(enforce_csrf_checks=True)
+        assert client.login(username='del@icio.us', password='password')
+        response = client.get(reverse('devhub.api_creds'), {'key': 'foobar'})
+
+        assert response.status_code == 404
+
```

Comment:
this needs another test like `test_cannot_get_other_users_secret()` which logs in as a different user but requests an otherwise valid key like `'key': self.api_key.key`. This should fail when the key is not owned by the user who is signed in.